gobblin-core/src/main/java/gobblin/source/extractor/extract/kafka/KafkaWrapper.java

```diff
@@ -0,0 +1,462 @@
+/* (c) 2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.source.extractor.extract.kafka;
+
+import gobblin.configuration.State;
+import gobblin.source.extractor.extract.EventBasedSource;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import kafka.api.PartitionFetchInfo;
+import kafka.api.PartitionOffsetRequestInfo;
+import kafka.common.TopicAndPartition;
+import kafka.javaapi.FetchRequest;
+import kafka.javaapi.FetchResponse;
+import kafka.javaapi.OffsetRequest;
+import kafka.javaapi.OffsetResponse;
+import kafka.javaapi.PartitionMetadata;
+import kafka.javaapi.TopicMetadata;
+import kafka.javaapi.TopicMetadataRequest;
+import kafka.javaapi.consumer.SimpleConsumer;
+import kafka.javaapi.message.ByteBufferMessageSet;
+import kafka.message.MessageAndOffset;
+
+import org.apache.commons.lang.NotImplementedException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Splitter;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+
+
+/**
+ * Wrapper class that contains two alternative Kakfa APIs: an old low-level Scala-based API, and a new API.
+ * The new API has not been implemented since it's not ready to be open sourced.
+ *
+ * @author ziliu
+ */
+public class KafkaWrapper implements Closeable {
+
+  private static final Logger LOG = LoggerFactory.getLogger(KafkaWrapper.class);
+
+  private static final String USE_NEW_KAFKA_API = "use.new.kafka.api";
+  private static final boolean DEFAULT_USE_NEW_KAFKA_API = false;
+  private static final String KAFKA_BROKERS = "kafka.brokers";
+
+  private final List<String> brokers;
+  private final KafkaAPI kafkaAPI;
+
+  private final boolean useNewKafkaAPI;
```

Comment:
Is there a cleaner way of doing this? Probably by just using an ENUM? What if there are more Kafka APIs that we want to add support for.